source/scpd/tests/GlueTypes.d

```diff
@@ -0,0 +1,74 @@
+
+/*******************************************************************************
+
+    Contains runtime size & ABI object layout checks and
+    function argument passing in the D glue layer works properly.
+
+    Copyright:
+        Copyright (c) 2019 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module scpd.tests.GlueTypes;
+
+import core.stdc.stdint;
+import std.meta;
+
+import scpd.types.Stellar_SCP;
+import scpd.types.Stellar_types;
+import scpd.types.Utils;
+import scpd.types.XDRBase;
+
+// todo: replace with __allMembers tricks in the future
+alias TypesWithLayout = AliasSeq!
+(
+    /// scpd.types.Stellar_SCP
+    SCPBallot,
+    SCPNomination,
+    SCPStatement._pledges_t._prepare_t,
+    SCPStatement._pledges_t._confirm_t,
+    SCPStatement._pledges_t._externalize_t,
+    SCPStatement._pledges_t,
+    SCPStatement,
+    SCPEnvelope,
+    SCPQuorumSet,
+
+    /// scpd.types.Stellar_types
+    //Hash,  // todo (linker error)
```

Comment:
`Hash` is aliased to `BitBlob!256`
But I would double check if they are binary compatible (esp. w.r.t. endianess)

Reply:
Ah I could just pass it as a void* then. Yeah I'll add those checks as well.

Reply:
Or you could pass it as a `ref Hash` on D side and `Hash&` on the C++ side?